### ISSUE TITLE
Correction is needed for the specification of subroutine arguments

### ISSUE THREAD

**Issue by hhorii** [Qiskit/openqasm#279]
### What is the expected enhancement?

[The current live specification](https://qiskit.github.io/openqasm/language/subroutines.html) says:

> Using subroutines, we can define an X-basis measurement with the program `def xmeasure qubit q -> bit { h q; return measure q; }`. We can also define more general classes of single-qubit measurements as `def pmeasure(angle[32] theta) qubit q -> bit { rz(theta) q; h q; return measure q; }`. The type declarations are necessary if we want to mix qubit and register arguments. For example, we might define a parity check subroutine that takes qubits and registers
```
def xcheck qubit[4] d, qubit a -> bit {
  reset a;
  for i in [0: 3] cx d[i], a;
  return measure a;
}
```

However, the paper (https://arxiv.org/pdf/2104.14722.pdf) uses the following code sample:

```
def segment(qubit[2] anc, qubit psi) -> bit[2] {
  bit[2] b;
  ...
}
```

[The current ANTLR grammar](https://github.com/Qiskit/openqasm/blob/fe80e627ad7b8bedbb4572058b5d375ef7f9b17e/source/grammar/qasm3.g4#L478-L481) follows the manner of the paper: 
```
subroutineDefinition
    : 'def' Identifier LPAREN anyTypeArgumentList? RPAREN
    returnSignature? subroutineBlock
    ;
```

I believe that the live specification needs correction about definition of subroutine arguments, as follows.
```
def xcheck (qubit[4] d, qubit a) -> bit {
  reset a;
  for i in [0: 3] cx d[i], a;
  return measure a;
}
```

**Comment by jakelishman** [comment]
Agree, I think this is just a mistake in the specification text here.

**Comment by jakelishman** [comment]
Closed by #285
